core-client/src/main/java/org/glassfish/jersey/client/authentication/DigestAuthenticator.java

```diff
@@ -349,6 +352,25 @@ private String randomBytes(final int nbBytes) {
         return bytesToHex(bytes);
     }
 
+    private static void discardInputAndClose(InputStream is) {
```

Comment:
Do we really need to consume the stream before it is closed?

Reply:
Consuming the stream makes it possible for the connection to be reused (unless the server closes the connection, of course).